packages/docusaurus-theme-classic/src/theme/AnnouncementBar/styles.module.css

```diff
@@ -22,51 +22,40 @@ html[data-announcement-bar-initially-dismissed='true'] .announcementBar {
   display: none;
 }
 
-@media print {
-  .announcementBar {
-    display: none;
-  }
-}
-
-@media screen and (min-width: 1024px) {
-  :root {
-    --docusaurus-announcement-bar-height: 30px;
-  }
-}
-
 .announcementBarClose {
-  position: absolute;
-  right: 0;
-  top: 0;
-  width: 55px;
+  flex: 0 0 10%;
   font-size: 1.25rem;
-  height: 100%;
 }
 
 .announcementBarContent {
+  flex: 1 1 auto;
   font-size: 85%;
-  width: 100%;
   text-align: center;
   padding: 5px 0;
 }
 
-.announcementBarCloseable {
-  margin-right: 55px;
-}
-
-@media screen and (max-width: 576px) {
-  .announcementBarClose {
-    width: 35px;
-  }
-  .announcementBarContent {
-    width: auto;
-  }
-  .announcementBarCloseable {
-    margin-right: 35px;
+@media print {
+  .announcementBar {
+    display: none;
   }
 }
 
 .announcementBarContent a {
   color: inherit;
   text-decoration: underline;
 }
+
+@media screen and (min-width: 1024px) {
+  :root {
+    --docusaurus-announcement-bar-height: 30px;
+  }
+
+  /* Adjust line-height for proper vertical centering on desktop */
```

Comment:
The placeholder could be an empty div of the same width as the close button that would just push the main content to the center.

Reply:
That looks nice overall but the content is not perfectly centered. It's not very visible, but if you increase the close button % width to something more significant, the centering issue is more visible, and this becomes more important to add a placeholder container on the other side.

![image](https://user-images.githubusercontent.com/749374/131313336-e9b7738a-ea00-459d-a1b3-a4125a2300e2.png)

Also, what about not using % for close button width?
At least for very small viewports, 4% is too small and doesn't really make sense to me, and the close button can touch the edges of the screen.

![image](https://user-images.githubusercontent.com/749374/131313778-0bcf1b66-4a99-4cb7-b546-2d5abbed2ecd.png)

Reply:
Clear now, just did as you suggested. But on mobiles with small screen, we see this:

![image](https://user-images.githubusercontent.com/4408379/131318896-bb4ca5d1-e848-401b-8b3e-3f57f451b554.png)


It's normal?

Reply:
I think it would look better if the placeholder used `flex-basis: 10px` for small widths? 
Centering is less important for mobile devices where we want to use more space

Reply:
> I think it would look better if the placeholder used flex-basis: 10px for small widths?

It doesn't look good, maybe just don't use the placeholder column on mobiles?

![image](https://user-images.githubusercontent.com/4408379/131323470-59fd8664-8e42-44dc-ab24-51eb9857d299.png)

Reply:
> It doesn't look good, maybe just don't use the placeholder column on mobiles?

This is not the result I get 🤪 Only the placeholder should have flex-basis 10 for small screens/mobile, not the close button 

![image](https://user-images.githubusercontent.com/749374/131328847-8ba0aa5d-b350-4517-b116-88d7a9d2ac94.png)

Reply:
> maybe just don't use the placeholder column on mobiles?

I'm fine with removing it too, as long as there's some padding that prevents the main content from touching the edge of the screen (which would be the same as applying that padding by setting a width to this placeholder)

Reply:
Ok, done it.